File: typescript/src/data_models/external_task/results/external_task_system_error.ts
import {ExternalTaskResultBase} from './external_task_result_base';

/**
 * Contains the result set for an ExternalTask that failed with a System error.
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export class ExternalTaskSystemError<TDetails = any> extends ExternalTaskResultBase {

  public readonly errorMessage: string;
  public readonly errorDetails: TDetails;

  constructor(externalTaskId: string, errorMessage: string, errorDetails: TDetails) {
    super(externalTaskId);
    this.errorMessage = errorMessage;
    this.errorDetails = errorDetails;
  }

}

File: dotnet/src/DataModels/ExternalTask/Results/ExternalTaskSystemError.cs
namespace ProcessEngine.ConsumerAPI.Contracts.DataModel
{
    /// <summary>
    /// Contains the result set for an ExternalTask that failed with a System error.
    /// </summary>
    public class ExternalTaskSystemError<TDetails> : ExternalTaskResultBase
    {
        /// <summary></summary>
        /// <param name="externalTaskId"></param>
        /// <param name="errorMessage"></param>
        /// <param name="errorDetails"></param>
        public ExternalTaskSystemError(string externalTaskId, string errorMessage, TDetails errorDetails)
            : base(externalTaskId)
        {
            this.errorMessage = errorMessage;
            this.errorDetails = errorDetails;
        }

        /// <summary>
        /// The mesage of the error that occured.
        /// </summary>
        /// <value></value>
        public string errorMessage { get; private set; }

        /// <summary>
        /// The details provided for this error.
        /// </summary>
        /// <value></value>
        public TDetails errorDetails { get; private set; }
    }
}
